tests/pytest/eligibility/test_views.py

```diff
@@ -71,6 +72,39 @@ def model_EnrollmentFlow_with_form_class(mocker, model_EnrollmentFlow):
     return model_EnrollmentFlow
 
 
+@pytest.mark.django_db
+def test_eligibility_logged_in_filtering_flows(mocker, model_TransitAgency, client):
+    model_TransitAgency.pk = None
+    model_TransitAgency.save()
+    digital = models.EnrollmentFlow.objects.create(
+        transit_agency=model_TransitAgency,
+        supported_enrollment_methods=[models.EnrollmentMethods.DIGITAL],
+        label="Digital",
+        selection_label_template="eligibility/includes/selection-label.html",
+    )
+    in_person = models.EnrollmentFlow.objects.create(
+        transit_agency=model_TransitAgency,
+        supported_enrollment_methods=[models.EnrollmentMethods.IN_PERSON],
+        label="In-Person",
+        selection_label_template="eligibility/includes/selection-label.html",
+    )
+    both = models.EnrollmentFlow.objects.create(
+        transit_agency=model_TransitAgency,
+        supported_enrollment_methods=[models.EnrollmentMethods.DIGITAL, models.EnrollmentMethods.IN_PERSON],
+        label="Both",
+        selection_label_template="eligibility/includes/selection-label.html",
+    )
+    model_TransitAgency.save()
+    model_TransitAgency.enrollment_flows.add(digital, in_person, both)
+    mocker.patch("benefits.core.session.agency", autospec=True, return_value=model_TransitAgency)
+
+    path = reverse(routes.ELIGIBILITY_INDEX)
+    response = client.get(path)
+
+    assert model_TransitAgency.enrollment_flows.count() == 3
+    assert len(response.context_data["form"].fields["flow"].choices) == 2
```

Comment:
> I would have liked to have been able to have an assertion on the last line of these specs that are more like "The response's form should have a field that is supported by digital", but the `context_data["form"].fields["flow"].choices` only returns an id of the flow, not name of the flow or anything more. So I decided to test it like this instead. I'm open to other ideas on how to test this!
(_from https://github.com/cal-itp/benefits/pull/2399#discussion_r1777666206)_)

I think we could do some assertions on the IDs.

You could get a list of the flow IDs with a list comprehension:
```python
flow_ids = [choice[0] for choice in response.context_data["form"].fields["flow"].choices]
```
and then check if `digital.id` and `both.id` are in that list. And check that `in_person.id` is not.